EventListener/JmsPreSerializeListener.php

```diff
@@ -44,13 +52,23 @@ public function onPreSerialize(ObjectEvent $event)
             foreach ($reflectionClass->getProperties() as $property) {
                 $liipAnnotation = $this->annotationReader->getPropertyAnnotation($property, LiipImagineSerializableField::class);
                 $property->setAccessible(true);
-
                 if ($liipAnnotation instanceof LiipImagineSerializableField && ($value = $property->getValue($object)) && !is_array($value)) {
                     $vichField = $liipAnnotation->getVichUploaderField();
 
+                    $uriComponents = explode($value, '/');
+                    $cacheKey      = $vichField.array_pop($uriComponents);
+
+                    if (in_array($cacheKey, $this->cache)) {
```

Comment:
I think we will have problem with cacheKey. For example if image with same name but from different folders come.

Reply:
In fact, didn't think about it. I should take the vich mapping attribute in the key too. Don't know how I can achieve that, I'll take a look